### PR TITLE
HDDS-10202. OmMetadataManagerImpl creates new S3Batcher for each operation

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Avoid `OmMetadataManagerImpl` creating new instance of `S3Batcher` for each S3 secret operation.

https://issues.apache.org/jira/browse/HDDS-10202

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/7643348101

### CODE REVIEW

**Comment by adoroszlai** [comment]
@Pochatkin please review

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java [hunk 1]
```diff
@@ -318,6 +318,24 @@ public class OmMetadataManagerImpl implements OMMetadataManager,
   private final Map<String, TableCacheMetrics> tableCacheMetricsMap =
       new HashMap<>();
   private SnapshotChainManager snapshotChainManager;
+  private final S3Batcher s3Batcher = new S3Batcher() {
```
Comment (swamirishi) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:321]: nit: How about having this object inside a constructor keeping this in a method like initializeS3Batcher()?

**Review by swamirishi — CHANGES_REQUESTED** [review]
@adoroszlai Thanks for the patch, overall the patch looks good to me. Had a minor nitpicky comment, Otherwise I am fine with the entire change.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java (hunk 1 above)
Comment (adoroszlai) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:321]: There are 4 constructors.  What is the benefit of duplicating the assignment in all of them?

Comment (adoroszlai) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java]: Converted the anonymous class to an inner one.

**Review by swamirishi — APPROVED** [review]
LGTM. Thanks for the patch @adoroszlai

**Comment by adoroszlai** [comment]
Thanks @Pochatkin, @swamirishi for the review.

> Thanks for fix my mistake.

No problem.

**Pull request merged**